libraries/chain/controller.cpp

```diff
@@ -3339,9 +3382,18 @@ struct controller_impl {
       }
 
       guard_pending.cancel();
+
       return onblock_trace;
    } /// start_block
 
+   void run_readonly_transactions() {
+      // update peer public keys from chainbase db using a readonly trx
+      auto block_num = chain_head.block_num();
+      if (block_num % 120 == 0) {
+         peer_keys_db.update_peer_keys(get_top_producer_keys()); // update once/minute
```

Comment:
You need a try-catch and drop here for if the system contract does not support the read-only trx.

Reply:
It seems that `push_transaction` silently fails and returns an empty return value when the action is not present.

Reply:
https://github.com/AntelopeIO/spring/pull/1316/commits/9c783fd7036de6e1efc44b64fd6d95df9e9a7a9a